bin/bump

```diff
@@ -1,69 +1,31 @@
 #! /usr/bin/php
 <?php
 
-$major = 0;
-$minor = 0;
-$patch = 0;
-
 if (empty($argv[1]) || !in_array($argv[1], ['major', 'minor', 'patch'])) {
-	echo 'Please supply a bump type: major, minor, patch.' . PHP_EOL;
-	exit;
+    echo 'Please supply a bump type: major, minor, patch.' . PHP_EOL;
+    exit;
 }
 
 $bump = $argv[1];
 
-$tags = explode(PHP_EOL, shell_exec('git tag'));
-foreach ($tags as $tag) {
-	$parts = explode('.', $tag);
-	if (count($parts) !== 3) {
-		continue;
-	}
-
-	if ($parts[0] > $major) {
-		$major = $parts[0];
-		$minor = $parts[1];
-		$patch = $parts[2];
-	} elseif ($parts[0] == $major) {
-		if ($parts[1] > $minor) {
-			$minor = $parts[1];
-			$patch = $parts[2];
-		} elseif ($parts[1] == $minor && $parts[2] > $patch) {
-			$patch = $parts[2];
-		}
-	}
-}
-
-$current_version = implode('.', [$major, $minor, $patch]);
+$comparison = new \Bump\Compare(function ($tag) {
```

Comment:
It'd be nice if there was some way to set this up(service provider) out side of bump, oh well though :(.